src/document/utils.js

```diff
@@ -161,16 +161,20 @@ function mapDoc(doc, cb) {
 function findInDoc(doc, fn, defaultValue) {
   let result = defaultValue;
   let hasStopped = false;
+
   function findInDocOnEnterFn(doc) {
+    if (hasStopped) {
+      // Stop processing rest docs
```

Comment:
Yes, the top seems to be a more correct place. Let's just not use the word "stop" in the comment:
```suggestion
      // Skip further processing
```